### ISSUE TITLE
asciidoctor-pdf not accessible

### ISSUE THREAD

**Issue by PierreR** [NixOS/nixpkgs#24293]
## Issue description
After installing `asciidoctor` in `17.03`, the executable `asciidoctor-pdf` is not accessible anymore. Trying to use `asciidoctor -b pdf won't work either

### Steps to reproduce
`nix-env -i asciidoctor` 
-> asciidoctor-pdf -> not available

```
→ asciidoctor -b pdf README.adoc 
asciidoctor: FAILED: missing converter for backend 'pdf'. Processing aborted. (RuntimeError)
```

## Technical details

* System: 17.03beta582.1e8c017 (Gorilla)
* Nix version: nix-env (Nix) 1.11.8
* Nixpkgs version: 17.03beta675.3a368d7

**Comment by globin** [comment]
cc @gpyh

**Comment by globin** [comment]
And ping @zimbatm who was working on ruby infrastructure last IIRC.

**Comment by 8573** [comment]
`asciidoctor -r asciidoctor-pdf -b pdf <AsciiDoc file>` works for me
in `nix-shell -p asciidoctor`, though I don't know whether it's
supposed not to need the `-r asciidoctor-pdf`.

**Comment by gpyh** [comment]
I have no idea why it does this. I'm going to investigate when I find some time.

**Comment by zimbatm** [comment]
It's probably a change in behavior of asciidoctor, which has been upgraded, where the extension has to be loaded explicitly now. Either use `asciidoctor -r asciidoctor-pdf -b pdf` or `asciidoctor-pdf` instead.

**Comment by globin** [comment]
@zimbatm but `asciidoctor-pdf` does not make it into `$out` only in `${ruby.gemPath}/bin` so it is not really accessible.

**Comment by fidergo-stephane-gourichon** [comment]
# Context

Hello. I'm here considering a migration from python asciidoc that already has custom PDF style, vector-rendered graphviz diagrams in HTML and PDF.

Installed through ruby gems like this:

    gem install --user-install asciidoctor
    gem install --user-install asciidoctor-diagram
    gem install --user-install asciidoctor-pdf --pre

# Same quirks here

After trying a "hello world" document with default backend, since `asciidoctor --help` mentions `-b`, it's natural to do:

    asciidoctor -b pdf asciidoctor_diagram.asciidoc.txt

Alas this fails.
Finding this very issue on github allows to find these working alternatives:

    asciidoctor-pdf asciidoctor_diagram.asciidoc.txt
    asciidoctor -r asciidoctor-pdf -b pdf asciidoctor_diagram.asciidoc.txt

# What's the actual bug?

Is the first failure intended behavior?

If yes, the doc should be fixed.  I can provide patch and pull request.
If no, then the program should be fixed.

Fixing such gotchas should help other people following a similar path.

# Additional information

`asciidoctor --help` says "additional backends are supported via extensions (e.g., pdf, latex)".  Is this actually meant to be a hint to use `-r`?  Perhaps it could be made more explicit, like "add `-r asciidoctor-pdf` to command line" or similar. If that's the way to go, I can provide patch and pull request.

Thank you for your attention.

**Comment by zimbatm** [comment]
wonderful report @fidergo-stephane-gourichon. inspired by your effort I created a fix at #34763
